connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnectorTest.java

```diff
@@ -123,13 +127,21 @@ public void testFindConsumerGroups() throws Exception {
         Collection<ConsumerGroupListing> groups = Arrays.asList(
                 new ConsumerGroupListing("g1", true),
                 new ConsumerGroupListing("g2", false));
+        Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>();
+        offsets.put(new TopicPartition("t1", 0), new OffsetAndMetadata(0));
         doReturn(groups).when(connector).listConsumerGroups();
-        doReturn(true).when(connector).shouldReplicate(anyString());
+        doReturn(true).when(connector).shouldReplicateByTopicFilter(anyString());
+        doReturn(true).when(connector).shouldReplicateByGroupFilter(anyString());
+        doReturn(offsets).when(connector).listConsumerGroupOffsets(anyString());
         List<String> groupFound = connector.findConsumerGroups();
 
         Set<String> expectedGroups = groups.stream().map(ConsumerGroupListing::groupId).collect(Collectors.toSet());
         assertEquals(expectedGroups, new HashSet<>(groupFound),
                 "Expected groups are not the same as findConsumerGroups");
+
+        doReturn(false).when(connector).shouldReplicateByTopicFilter(anyString());
+        List<String> topicFilterGroupFound = connector.findConsumerGroups();
+        assertTrue(topicFilterGroupFound.size() == 0);
```

Comment:
Nit: we can improve failure messages by making a more general comparison here:
```suggestion
        assertEquals(Collections.emptyList(), topicFilterGroupFound);
```

Reply:
nice!